ci/xcode

```diff
@@ -0,0 +1,50 @@
+name: Xcode
+
+on: [push]
+
+jobs:
+  build:
+  
+    name: Build and Test ${{ matrix.scheme }} for ${{ matrix.device }}
+    runs-on: macOS-latest
+    
+    strategy:
+      max-parallel: 6
+      matrix:
+        # List all schemes to build
+        scheme: [default]
+        # Specify the platform & devices
+        platform: [iOS Simulator]
+        device: [iPhone Xs, iPhone Xs Max]
+    
+    steps:
+    
+    - uses: actions/checkout@v1
+    
+      # Runs if matrix.scheme is set to default
+      # Finds the first listed scheme
+    - name: Set Default Scheme
+      if: matrix.scheme == 'default'
+      run: |
+        scheme_list=$(xcodebuild -list -json | tr -d "\n")
+        default=$(echo $scheme_list | ruby -e "require 'json'; puts JSON.parse(STDIN.gets)['project']['targets'][0]")
```

Comment:
this will not work with straight SwiftPM projects, the JSON changes to something like: 
```JSON
{"workspace" : {"name" : "THOTP", "schemes" : ["THOTP"]}}
```

Reply:
From my understanding, this should still work for Xcode applications other than just a SwiftPM project?

Should that be incorporated? Or does that belong in a separate workflow?

I don't want this workflow to get too incomprehensible.

There is a PR open specifically for SwiftPM: #21

Reply:
Check out the PR i just posted this week, my approach is a lot simpler than yours but includes support to force Xcode into version 11.1 (default on the github runner is to use 10.3), which might be why Xcode SwiftPM doesn't work.  I did not use matrix schemes or ruby scripts so my approach is a little less flexible but maybe easier to incorporate for some people. 

My current project loads Lottie via SwiftPM and it builds on my GitHub Action CI well.

Reply:
> this will not work with straight SwiftPM projects, the JSON changes to something like: 
> 
> ```JSON
> 
> {"workspace" : {"name" : "THOTP", "schemes" : ["THOTP"]}}
> 
> ```